### PR TITLE
[OSC-1238] Add CI/CD

### PR DESCRIPTION
**Pull request by waltaro**
# OSC-1238
### Overview
- Integration with Travis CI! 🎉 
- Add [Pipenv](https://pipenv.readthedocs.io/en/latest/) support (highly recommended to manage dependencies/venv)
- Add [PyTest](https://docs.pytest.org/en/latest/) for when we do write unit tests

### Changes
- Add build status to `README.md`
- Add `.travis.yml`
- Add `Pipfile` and `Pipfile.lock`
- Add `tests` folder directory
    - Contains `integration` and `unit` tests
- Rename `psycopg2` dependency to `psycopg2-binary`
    - Update `setup.py` to reflect changes

#### CI/CD
Travis builds will perform two types of testing:
- Unit testing
- Integration testing

Integration tests perform the following:

1. Create stub load models
    a. load_model_1.json, load_model_2.json
2. Initialise a new execution
3. Compare load models and assert that the models changed are `*` (as no models have been compared)
4. Complete execution and assert that last successful execution ID is the one returned from `init`
5. Initalise another execution
6. Modify `load_model_1.json`
7. Compare load models and assert that models changed are `load_model_1.json`
8. Complete execution and assert that last successful execution ID is the one returned from `init`

### Readings
- [Why you should use pyenv + Pipenv for your Python projects](https://hackernoon.com/reaching-python-development-nirvana-bb5692adf30c)
- [What is a Makefile and how does it work?](https://opensource.com/article/18/8/what-how-makefile)
- [Testing Projects](https://github.com/pypa/pipenv/blob/master/docs/advanced.rst#-testing-projects)
- [Requests](https://github.com/kennethreitz/requests)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 sqlalchemy==1.2.17
-psycopg2==2.7.7
+psycopg2-binary==2.7.7
```
Comment (ChintanRaval) [inline requirements.txt]: ensure corresponding changes are also made to [setup.py](https://github.com/PageUpPeopleOrg/model-change-detector/blob/feature/OSC-1238_ci-cd/setup.py)

Comment (waltaro) [reply requirements.txt]: Great pickup! Thank you!

**Comment by ChintanRaval** [comment]
Love the PR description.. very thorough 👍

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	pip install pipenv --upgrade
+	pipenv install --dev
+
+# Run unit tests
+test_unit:
+	pipenv run pytest
+
+# Run integration tests
+test_integration:
+	./tests/integration/test_integration.sh
```
Comment (seanbudd) [inline Makefile]:
```suggestion
	./tests/integration/test_integration.sh
	
```

File: README.md [hunk 3]
```diff
@@ -1,20 +1,13 @@
...
-```commandline
-py mcd.py [options] <command> [command-parameters]
 ```
-
-To get help,use:
-
-```commandline
-py mcd.py -h
-py mcd.py <db-connection-string> <command> -h
+$ python mcd.py [options] <command> [command-parameters]
```
Comment (seanbudd) [inline README.md]: is the db-connection-string no longer required before the command?

**Pull request merged**